wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt

```diff
@@ -0,0 +1,111 @@
+ *
+ * @param stack a [ChildStack] to be displayed.
+ * @param onDismissed called when the swipe to dismiss gesture has completed, allows popping the stack.
+ * See [StackNavigator#pop][com.arkivanov.decompose.router.stack.pop].
+ * @param modifier a [Modifier] to be applied to the underlying
+ * [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ * @param content a Composable slot displaying a [Child][Child.Created].
+ */
+@Composable
+fun <C : Any, T : Any> SwipeToDismissBox(
+    stack: Value<ChildStack<C, T>>,
+    onDismissed: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable (child: Child.Created<C, T>) -> Unit,
+) {
+    val state = stack.subscribeAsState()
+
+    SwipeToDismissBox(
+        stack = state.value,
+        onDismissed = onDismissed,
+        modifier = modifier,
+        content = content,
+    )
+}
+
+/**
+ * Displays the [ChildStack] in [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ *
+ * @param stack a [ChildStack] to be displayed.
+ * @param onDismissed called when the swipe to dismiss gesture has completed, allows popping the stack.
+ * See [StackNavigator#pop][com.arkivanov.decompose.router.stack.pop].
+ * @param modifier a [Modifier] to be applied to the underlying
+ * [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ * @param content a Composable slot displaying a [Child][Child.Created].
+ */
+@Composable
+fun <C : Any, T : Any> SwipeToDismissBox(
+    stack: ChildStack<C, T>,
+    onDismissed: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable (child: Child.Created<C, T>) -> Unit,
+) {
+    val active: Child.Created<C, T> = stack.active
+    val background: Child.Created<C, T>? = stack.backStack.lastOrNull()
+    val holder = rememberSaveableStateHolder()
+
+    RetainStates(holder, stack.getConfigurations())
+
+    SwipeToDismissBox(
+        onDismissed = onDismissed,
+        modifier = modifier,
+        backgroundScrimColor = workaroundBackgroundScrimColor(stackSize = stack.items.size),
+        backgroundKey = background?.configuration ?: SwipeToDismissKeys.Background,
+        contentKey = active.configuration,
+        hasBackground = background != null,
+    ) { isBackground ->
+        val child = background?.takeIf { isBackground } ?: active
+        holder.SaveableStateProvider(child.configuration.key()) {
+            content(child)
+        }
+    }
+}
+
+// Workaround for https://issuetracker.google.com/issues/280392104
+@Composable
+private fun workaroundBackgroundScrimColor(stackSize: Int): Color {
+    return MaterialTheme.colors.background.copy(alpha = 1F - ((stackSize % 2) * 0.01F))
+}
+
+private fun ChildStack<*, *>.getConfigurations(): Set<String> =
+    items.mapTo(HashSet()) { it.configuration.key() }
+
+private fun Any.key(): String = "${this::class.simpleName}_${hashCode().toString(radix = 36)}"
```

Comment:
This looks a bit unexplained but from https://github.com/arkivanov/Decompose/blob/6fa7b41d6f3050f47194d5431eb16a3b4c6e7c10/extensions-compose-jetpack/src/main/java/com/arkivanov/decompose/extensions/compose/jetpack/stack/Children.kt#L57

Reply:
Thanks for the review, I will add a comment for this.